server/route.js

```diff
@@ -1,195 +1,196 @@
+      $('body').append(heads);
+  
+      // Remove empty lines caused by removing scripts
+      $('head').html($('head').html().replace(/(^[ \t]*\n)/gm, ''));
+  
+      return $.html();
+    }
   }
-};
-
-Route.prototype._processFromCache = function(pageInfo, res, next) {
-  var originalWrite = res.write;
-  res.write = function(data) {
-    originalWrite.call(this, pageInfo.html);
+  
+  _processFromCache(pageInfo, res, next) {
+    const originalWrite = res.write;
+    res.write = (data) => {
+      originalWrite.call(this, pageInfo.html);
+    }
+  
+    res.pushData('fast-render-data', pageInfo.frData);
+    next();
   }
-
-  res.pushData('fast-render-data', pageInfo.frData);
-  next();
-};
-
-Route.prototype._buildContext = function(url, params, queryParams) {
-  var context = {
-    route: this,
-    path: url,
-    params: params,
-    queryParams: queryParams
-  };
-
-  return context;
-};
-
-Route.prototype.isHtmlPage = function(url) {
-  var pathname = Url.parse(url).pathname;
-  var ext = pathname.split('.').slice(1).join('.');
-
-  // if there is no extention, yes that's a html page
-  if(!ext) {
-    return true;
+  
+  _buildContext(url, params, queryParams) {
+    const context = {
+      route: this,
+      path: url,
+      params: params,
+      queryParams: queryParams
+    };
+  
+    return context;
   }
-
-  // if this is htm or html, yes that's a html page
-  if(/^htm/.test(ext)) {
-    return true;
+  
+  isHtmlPage(url) {
+    const pathname = Url.parse(url).pathname;
+    const ext = pathname.split('.').slice(1).join('.');
+  
+    // if there is no extention, yes that's a html page
+    if(!ext) {
+      return true;
+    }
+  
+    // if this is htm or html, yes that's a html page
+    if(/^htm/.test(ext)) {
+      return true;
+    }
+  
+    // if not we assume this is not as a html page
+    // this doesn't do any harm. But no SSR
+    return false;
   }
-
-  // if not we assume this is not as a html page
-  // this doesn't do any harm. But no SSR
-  return false;
-};
-
-Route.prototype._lookupCachedPage= function(url) {
-  var info = this._cache[url];
-  if(info) {
-    return info.data;
+  
+  _lookupCachedPage(url) {
+    const info = this._cache[url];
+    if(info) {
+      return info.data;
+    }
   }
-};
-
-Route.prototype._cachePage = function(url, data, timeout) {
-  var self = this;
-  var existingInfo = this._cache[url];
-  if(existingInfo) {
-    clearTimeout(existingInfo.timeoutHandle);
+  
+  _cachePage(url, data, timeout) {
+    const existingInfo = this._cache[url];
+    if(existingInfo) {
+      clearTimeout(existingInfo.timeoutHandle);
+    }
+  
+    const info = {
+      data: data,
+      timeoutHandle: setTimeout(() => {
+        delete this._cache[url];
+      }, timeout)
+    };
+  
+    this._cache[url] = info;
   }
-
-  var info = {
-    data: data,
-    timeoutHandle: setTimeout(function() {
-      delete self._cache[url];
-    }, timeout)
+  
+  register(name, sub, options) {
```

Comment:
You mean these 3?
```
  register(name, sub, options) {
    this._subsMap[name] = sub;
  };
  
  subscription(name) {
    return this._subsMap[name];
  }
  
  middleware(middleware) {
  }
```

In the demo app I was using it caused an error. This app: https://github.com/timbrandin/meteor-react-flowrouter-jsx-templating-example